src/core/RenderContext.cpp

```diff
@@ -130,17 +140,21 @@ void RenderContext::updateGLWindows()
 
     BOOST_FOREACH( WallWindowPtr window, windows_ )
     {
+        window->setBlockDrawCalls( false );
         window->viewport()->repaint();
+        window->setBlockDrawCalls( true );
         ++activeGLWindowIndex_;
     }
+    glFinish();
 }
 
 void RenderContext::swapBuffers()
 {
     BOOST_FOREACH( WallWindowPtr window, windows_ )
     {
-        static_cast<QGLWidget*>( window->viewport( ))->makeCurrent();
-        static_cast<QGLWidget*>( window->viewport( ))->swapBuffers();
+        QGLWidget* glContext = static_cast<QGLWidget*>( window->viewport( ));
+        glContext->makeCurrent();
+        glContext->swapBuffers();
```

Comment:
I don't know; It works perfectly fine without it, and if I understand correctly:
- glFlush() blocks until all the gl commands have finished executing
- It does not imply that it will speed up their execution.
If the above is correct, I don't see the need to add it...